server/pages/index.js

```diff
@@ -342,13 +342,84 @@ function setUpLoggedInRoute( req, res, next ) {
 	}
 }
 
-function setUpRoute( req, res, next ) {
-	if ( req.cookies.wordpress_logged_in ) {
-		// the user is probably logged in
-		setUpLoggedInRoute( req, res, next );
-	} else {
-		setUpLoggedOutRoute( req, res, next );
+/**
+ * Sets up a Content Security Policy header
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP
+ * @param {Object} req Express request object
+ * @param {Object} res Express response object
+ * @param {Function} next a callback to call when done
+ */
+function setUpCSP( req, res, next ) {
+	const originalUrlPathname = req.originalUrl.split( '?' )[ 0 ];
+
+	// We only setup CSP for /log-in* for now
+	if ( ! /^\/log-in/.test( originalUrlPathname ) ) {
+		next();
+		return;
 	}
+
+	// This is calculated by taking the contents of the script text from between the tags,
```

Comment:
I see, so could we generate it from `document/index.jsx` instead of using constants?